autonomi/src/networking/driver/task_handler.rs

```diff
@@ -132,35 +133,32 @@ impl TaskHandler {
                     PrettyPrintRecordKey::from(&record.record.key)
                 );
                 let holders = self.get_record_accumulator.entry(id).or_default();
+
                 if let Some(peer_id) = record.peer {
                     holders.insert(peer_id, record.record);
                 }
+
+                // If we have enough holders, finish the task.
+                if let Some((_resp, quorum)) = self.get_record.get(&id) {
+                    let expected_holders = get_quorum_amount(quorum);
+
+                    if holders.len() >= expected_holders {
```

Comment:
for the situation of mutable data having too many versions (or nasty nodes respond chunk with corrupted data),
such simple check of `number of holders` could introduce potential risk ?

Reply:
Would increasing the `Quorum` that can be passed to the GetRecord request mitigate that risk?

Reply:
>  we should return the record held by the majority of peers.

no, the `split` records shall always be returned

I just mean, in case of split, instead of `return on Quorum holders`, we shall `return on single version has Quorum holders` with all versions collected

> this should be handled in a separate PR

right, ok with leave this to separate PR, as I's also suggest to use req/rsp DM to replace kad::get_record